diff --git a/virtualcluster/pkg/controller/kubeconfig/kubeconfig.go b/virtualcluster/pkg/controller/kubeconfig/kubeconfig.go
index 5be3386..95e0934 100644
--- a/virtualcluster/pkg/controller/kubeconfig/kubeconfig.go
+++ b/virtualcluster/pkg/controller/kubeconfig/kubeconfig.go
@@ -86,7 +86,10 @@ func generateKubeconfigUseCertAndKey(clusterName string, ips []string, apiserver
 	urls := make([]string, 0, len(ips))
 	for _, ip := range ips {
 		addr := net.ParseIP(ip)
-		if addr.To4() != nil {
+		if addr == nil {
+			// is a hostname/domain name (not an IP address)
+			urls = append(urls, fmt.Sprintf("https://%v:6443", ip))
+		} else if addr.To4() != nil {
 			// is ipv4
 			urls = append(urls, fmt.Sprintf("https://%v:6443", ip))
 		} else {
